packages/callbacks/tizen/inc/macros.h

```diff
@@ -0,0 +1,401 @@
+#ifndef _TIZEN_INTEROP_CALLBACKS_MACROS_H_
+#define _TIZEN_INTEROP_CALLBACKS_MACROS_H_
+
+#define EXPAND_LIST(...) __VA_ARGS__
+
+#define GET_CALLBACK_ID(CB_NAME, OFFSET) \
+  __reserved_cb_id_array[BASE_CALLBACK_ID_##CB_NAME + OFFSET]
+
+// Note: CB_PARAMS_NAMES should be defined before calling PROXY_GROUP_*.
+
+#define PROXY_GROUP_RETURN(CB_NAME, CB_RETURN, CB_PARAMS...)                \
+  CB_RETURN platform_blocking_##CB_NAME(CB_PARAMS) {                        \
+    if (!user_data) {                                                       \
+      LOG_ERROR("Received null user_data in %s - aborting.", #CB_NAME);     \
+      return CB_RETURN();                                                   \
+    }                                                                       \
+    int callbackId = reinterpret_cast<int>(user_data);                      \
+    LDEBUG("enter <RETURN> id:%d", callbackId);                             \
+    CallbackInfo cbInfo = __cb_id_to_info_map[callbackId];                  \
+    user_data = cbInfo.actualUserData;                                      \
+    if (gettid() == main_thread_id) {                                       \
+      LDEBUG("calling dart callback directly");                             \
+      return (reinterpret_cast<CB_NAME>(cbInfo.callbackPtr))(EXPAND_LIST(   \
+          CB_PARAMS_NAMES));                                                \
```

Comment:
I've used that term in a few places, so I better correct that and replace with "callback handling thread" or "origin thread" or introduce term like "TizenInteropCallbacks thread"

Reply:
Okay, I don't really have a good suggestion for the name. Anyway if you are going to update the documentation, you may use the term "root isolate" instead of "main isolate" for general consistency.